cmd/kubeadm/app/cmd/phases/init/kubeletfinalize.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	kubeletphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet"
+)
+
+var (
+	kubeletFinalizePhaseExample = cmdutil.Examples(`
+		# Updates settings relevant to the kubelet after TLS bootstrap"
+		kubeadm init phase kubelet-finalize all --config
+		`)
+)
+
+// NewKubeletFinalizePhase creates a kubeadm workflow phase that updates settings
+// relevant to the kubelet after TLS bootstrap.
+func NewKubeletFinalizePhase() workflow.Phase {
+	return workflow.Phase{
+		Name:    "kubelet-finalize",
+		Short:   "Updates settings relevant to the kubelet after TLS bootstrap",
+		Example: kubeletFinalizePhaseExample,
+		Phases: []workflow.Phase{
+			{
+				Name:           "all",
+				Short:          "Run all kubelet-finalize phases",
+				InheritFlags:   []string{options.CfgPath, options.CertificatesDir},
+				Example:        kubeletFinalizePhaseExample,
+				RunAllSiblings: true,
+			},
+			{
+				Name:         "cert-rotation",
```

Comment:
We may as well prefix this with `experimental-` as well. If we decide to keep it we can rename it and add an alias to the old name. As long as all of the sub-phases of `kubelet-finalize` are marked as experimental, then that can be `experimental-` prefixed too.
It's best to decide as soon as possible if we want to merge this or not. I am in for merging as long as we can change things in the future (remove the phase and opt in for another solution).
@neolit123 @fabriziopandini WDYT?

Reply:
updating with experimental and also using EXPERIMENTAL in the description because it's better than alpha.

Reply:
the experimental- pattern seems fine in terms of giving us the option to remove things.
we have no other way...